src/components/avatar/Avatar.js

```diff
@@ -55,21 +61,28 @@ class Avatar extends React.PureComponent<Props, State> {
     };
 
     render() {
-        const { avatarUrl, className, name, id, size = '' }: Props = this.props;
+        const { avatarUrl, className, name, id, size = '', isPending = false }: Props = this.props;
         const { hasImageErrored }: State = this.state;
         const classes = classNames(['avatar', className, { [`avatar--${size}`]: SIZES[size] }]);
 
+        const hasNoAvatar = !avatarUrl && !isPending;
+        const loadingBackgroundColor = lightGray;
+        const initialsBackgroundColor = `${getColor(id)}`;
+        const backgroundColor = hasNoAvatar || hasImageErrored ? initialsBackgroundColor : loadingBackgroundColor;
+
         let avatar;
-        if (avatarUrl && !hasImageErrored) {
-            avatar = <AvatarImage onError={this.onImageError} url={avatarUrl} />;
-        } else if (name) {
-            avatar = <AvatarInitials id={id} name={name} />;
+        if (!isPending && avatarUrl && !hasImageErrored) {
+            avatar = <AvatarImage onError={this.onImageError} url={`${avatarUrl}`} />;
+        } else if (!isPending && name) {
+            avatar = <AvatarInitials name={name} />;
+        } else if (isPending) {
```

Comment:
Could this go first and then we could avoid the !isPending for the other checks?